plugins/commands/create.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+const validator = require('screwdriver-command-validator');
+const hoek = require('hoek');
+const urlLib = require('url');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/commands',
+    config: {
+        description: 'Create a new command',
+        notes: 'Create a specific command',
+        tags: ['api', 'commands'],
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => validator(request.payload.yaml)
+            .then((config) => {
+                if (config.errors.length > 0) {
+                    throw boom.badRequest(
+                        'Command has invalid format: ', config.errors.toString());
+                }
+
+                const commandFactory = request.server.app.commandFactory;
+
+                return Promise.all([
+                    commandFactory.list({
+                        params: {
+                            namespace: config.command.namespace,
+                            name: config.command.name
+                        }
+                    })
+                ]).then(([commands]) => {
+                    const commandConfig = hoek.applyToDefaults(config.command, {
+                        // TODO: implement in the phase 2 or later.
+                        // pipelineId: pipeline.id,
+                        // labels: config.command.labels || []
+                    });
+
+                    // If command name doesn't exist yet, just create a new entry
+                    if (commands.length === 0) {
+                        return commandFactory.create(commandConfig);
+                    }
+
```

Comment:
I think we should also add comment about checking permission here.

Reply:
Okay, I added some comments for the next development.
Thanks.